tests/rptest/tests/redpanda_test.py

```diff
@@ -64,6 +67,23 @@ def topic(self):
         assert len(self.topics) == 1
         return self.topics[0].name
 
+    @property
+    def debug_mode(self):
+        """
+        Useful for tests that want to change behaviour when running on
+        the much slower debug builds of redpanda, which generally cannot
+        keep up with significant quantities of data or partition counts.
+        """
+        return os.environ.get('BUILD_TYPE', None) == 'debug'
```

Comment:
it'd be interesting to see if we could derive this from the binary itself. but I guess as long as `task` and CI is a driver then it should be straight forward to maintain the environment context.